simphony/testing/abc_check_engine.py

```diff
@@ -212,6 +215,7 @@ def test_dataset_rename(self):
         # we should be able to access using the new "bar" name
         ds_bar = engine.get_dataset("bar")
         self.assertEqual(ds_bar.name, "bar")
+        engine.remove_dataset("bar")
```

Comment:
We can add the if statement there, but this test method it not really testing the number of datasets that are allowed to be stored, only renaming, so we could make it independent of ```number_datasets_used_in_testing```. Adding of more than one datasets is tested elsewhere for engines that support it....

Just wanted to underline this, but anyway, either way is good for me.